app/routes/_public.terms.tsx

```diff
@@ -1,17 +1,27 @@
+import { useSearchParams } from 'react-router';
 import logo from '~/assets/full_logo.png';
 import termsContent from '~/assets/terms-of-use.md?raw';
 import { Markdown } from '~/components/markdown';
 import { ScrollArea } from '~/components/ui/scroll-area';
+import { LinkWithViewTransition } from '~/lib/transitions';
 
 export default function TermsPage() {
+  const [searchParams] = useSearchParams();
+  const redirectTo = searchParams.get('redirectTo');
+
   return (
-    <ScrollArea className=" mx-auto h-dvh max-w-4xl px-4 py-8" hideScrollbar>
+    <ScrollArea className="mx-auto h-dvh max-w-4xl px-4 py-8" hideScrollbar>
       <header className="mb-8 flex items-center justify-start">
-        <a href="/">
-          <img src={logo} alt="Agicash Logo" className="mr-4 h-16" />
-        </a>
+        <LinkWithViewTransition
+          to={{
+            pathname: redirectTo ?? '/',
+          }}
+          transition="slideDown"
+          applyTo="oldView"
+        >
+          <img src={logo} alt="Agicash Logo" className="mr-4 h-8" />
```

Comment:
so clicking on logo takes you back? that seems odd. don't remember seeing that anywhere. usually logo click takes you home and for back there is back/x button